src/components/Gauge.js

```diff
@@ -76,119 +53,141 @@ const StatusGaugeWrapper = styled.div`
-    return carry + item.value
-  }, 0)
-
-  const graph = Pie({
-    accessor: item => item.value,
-    center: [0, 0],
-    data: data,
-    R: 46,
-    r: 25
-  })
-
-  const gauge = graph.curves
-  return (
-    <StatusGaugeWrapper {...props}>
-      <svg
-        className='gauge__graph'
-        viewBox='0 0 100 100'
-        xmlns='http://www.w3.org/2000/svg'
-      >
-        <circle
-          fill=' white'
-          stroke='#dadada'
-          strokeWidth='5'
-          cx='50'
-          cy='50'
-          r='46'
-        />
-
-        <g>
-          <circle fill={placeholderColor} cx='50' cy='50' r='46' />
-          <circle fill='white' cx='50' cy='50' r='25' />
-        </g>
-
-        <g transform='translate(50, 50)'>
-          {map(gauge, segment => {
-            if (!count) {
-              return
-            }
-            if (segment.item.value === count) {
-              return (
-                <g key={segment.item.key || segment.item.name}>
-                  <path
-                    transform='translate(-100, -100)'
-                    fill={segment.item.color}
-                    d='M100 100m-46,0a46,46,0 1,0 92,0a 46,46 0 1,0 -92,0zM100 100m-25,0a25,25,0 0,1 50,0a 25,25 0 0,1 -50,0z'
-                  />
-                </g>
-              )
-            }
+class StatusGauge extends React.Component {
+  componentDidMount () {
+    this.generateDoughnut(this.props)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.generateDoughnut(nextProps)
+  }
+
+  generateDoughnut (props) {
+    const doughnutRef = this.refs.doughnut
+
+    if (!doughnutRef) {
+      return false
+    }
+
+    // Redraw the chart on props update
+    if (DoughnutChart) {
+      DoughnutChart.destroy()
+    }
+
+    const ctx = doughnutRef.getContext('2d')
+
+    const { values, colors, labels, count } = this.normalizeProps(props.data)
+
+    const data = this.formatGraphData(values, colors, labels, count)
+    const options = this.formatGraphOptions(count)
+    const pluginOptions = this.formatPluginOptions(
+      doughnutDefaults,
+      font,
+      count
+    )
+
+    // eslint-disable-next-line
+    DoughnutChart = new Chart(ctx, {
+      type: 'doughnut',
+      data,
+      options,
+      plugins: [doughnutLabel],
+      config: {
+        plugins: {
+          'doughnut-label': pluginOptions
+        }
+      }
+    })
+  }
+
+  normalizeProps (data) {
+    const values = data.map(x => x.value)
+    const labels = data.map(x => x.name)
+    const colors = data.map(x => x.color)
+    const count = values.reduce((carry, item) => carry + item, 0)
+
+    return { values, labels, colors, count }
+  }
+
+  formatGraphData (values, colors, labels, count) {
+    return {
+      labels,
+      datasets: [
+        {
+          backgroundColor: count > 0 ? colors : this.props.placeholderColor,
+          data: count > 0 ? values : [1]
+        }
+      ]
+    }
+  }
+
+  formatGraphOptions (count) {
```

Comment:
How about `getGraphOptions` and `getLabelOptions`?
Although I would also be fine with combining those two into a single one.